File: backends/dotnet/src/Zappy.Application/Accounts/WishlistOwner.cs
namespace Zappy.Application;

public sealed class WishlistOwner(VisitorCart visitorCart)
{
    public async Task<string?> Find(Visitor visitor, CancellationToken cancellationToken) =>
        visitor.CustomerId ?? (await visitorCart.Find(visitor, cancellationToken))?.Id;

    public async Task<string> FindOrStartOne(Visitor visitor, CancellationToken cancellationToken) =>
        visitor.CustomerId ?? (await visitorCart.FindOrStartOne(visitor, cancellationToken)).Id;
}

File: backends/java/zappy-application/src/main/java/com/zappymart/application/accounts/WishlistOwner.java
package com.zappymart.application.accounts;

import com.zappymart.application.Visitor;

public final class WishlistOwner {

    private WishlistOwner() {
    }

    public static String of(Visitor visitor) {
        return visitor.isSignedIn() ? visitor.customerId() : visitor.cartId();
    }
}
